src/Storages/VirtualColumnUtils.cpp

```diff
@@ -115,7 +125,29 @@ NameSet getVirtualNamesForFileLikeStorage()
     return {"_path", "_file", "_size", "_time"};
 }
 
-VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription & storage_columns)
+std::unordered_map<std::string, std::string> parseHivePartitioningKeysAndValues(const String & path, const ColumnsDescription & storage_columns)
+{
+    std::string pattern = "([^/]+)=([^/]+)/";
```

Comment:
The first one should be not greedy, the second one should be greedy.
You can use `[^/=]`